nfv_sriov.md

```diff
@@ -0,0 +1,253 @@
+`OpenStackDataPlaneNodeSet`
+[EDPM services list](https://openstack-k8s-operators.github.io/dataplane-operator/composable_services)
+to replace the `nova` service with `nova-custom-sriov`.
+
+```yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneNodeSet
+spec:
+  ...
+  roles:
+    edpm-compute:
+      ...
+      services:
+        - bootstrap
+        - download-cache
+        - configure-network
+        - validate-network
+        - install-os
+        - configure-os
+        - ssh-known-hosts
+        - run-os
+        - reboot-os
+        - install-certs
+        - libvirt
+        - ovn
+        - neutron-ovn
+        - nova-custom-sriov
+        - neutron-sriov
+        - neutron-metadata
+```
+The `bootstrap`, `configure-network` and `neutron-sriov` services
+should be added before the `libvirt` and `nova` (or `nova-custom-sriov`
+in this case) services. It configures EDPM nodes with SRIOV.
+
+When the `nova-custom-sriov` service Ansible job runs, it will copy
+overrides from the `ConfigMap`s onto the Nova hosts.
+
+### Create the OpenStackDataPlaneNodeSet
+
+Create the CR from your directory based on the example
+[dataplane_v1beta1_openstackdataplanenodeset_sriov](https://github.com/openstack-k8s-operators/dataplane-operator/tree/main/examples/sriov)
+with the changes described in the previous section.
+```
+oc kustomize --load-restrictor LoadRestrictionsNone dataplane-operator/examples/sriov > dataplane_cr.yaml
+```
+
+### Create a OpenStackDataPlaneDeployment
+
+Creating an `OpenStackDataPlaneDeployment` will trigger Ansible jobs
+to configure an EDPM. Which Ansible roles are run depends on the
+`services` list.
+
+Each `OpenStackDataPlaneDeployment` can have its own
+`servicesOverride` list which will redefine the list
+of services of an `OpenStackDataPlaneNodeSet` for a
+deployment.
+
+The example
+[dataplane_v1beta1_openstackdataplanedeployment_sriov](https://github.com/openstack-k8s-operators/dataplane-operator/tree/main/examples/sriov)
+
+Create the CR based on the example
+```
+oc kustomize --load-restrictor LoadRestrictionsNone dataplane-operator/examples/sriov > dataplane_cr.yaml
+```
+
+Custom `OpenStackDataPlaneService` called `nova-custom-sriov`
+has been created as described in
+the [documentation to configure OpenStack to use SRIOV](sriov.md).
+The `nova-custom-sriov` can be seen in the
+[example](https://github.com/openstack-k8s-operators/dataplane-operator/tree/main/examples/sriov)
+and takes the place of the default `nova`
+OpenStackDataPlaneService. This custom service uses a `ConfigMap` called
+`cpu-pinning-nova` which ensures that the file `03-cpu-pinning-nova.conf` is used
+by Nova.
+
+Now that the `nova-custom-sriov` has been created, use the example
+[dataplane_v1beta1_openstackdataplanedeployment_sriov](https://github.com/openstack-k8s-operators/dataplane-operator/tree/main/examples/sriov)
+to start the second deployment.
```

Comment:
What is the second deployment?  Isn't there only one `OpenStackDataPlaneDeployment`?

Reply:
Nothing is preventing us to start multiple deployments.

Reply:
Yes, true.  But where is the need for a second deployment in the process being described here?

Reply:
only one deployment. removed it.